src/common/utxobased/engine/makeUtxoEngine.ts

```diff
@@ -132,6 +140,209 @@ export async function makeUtxoEngine(
   })
 
   const engine: EdgeCurrencyEngine = {
+    async accelerate(edgeTx: EdgeTransaction): Promise<EdgeTransaction | null> {
+      // Get the replaced transaction rom the processor:
+      const replacedTxid = edgeTx.txid
+      const [replacedTx] = await processor.fetchTransactions({
+        txId: replacedTxid
+      })
+
+      // Transaction checks:
+      // The transaction must be found and not confirmed or dropped.
+      if (replacedTx == null) return null
+      if (replacedTx.blockHeight !== 0) return null
+
+      // Double the fee used for the RBF transaction:
+      const vBytes = transactionSizeFromHex(replacedTx.hex)
+      const newFeeRate = Math.round(parseInt(replacedTx.fees) / vBytes) * 2
+
+      const replacedTxInputs = replacedTx.inputs
+      // Recreate UTXOs from processor transaction and mark them as unspent:
+      const replacedTxUtxos = await Promise.all(
+        replacedTxInputs.map(async (_, index) => {
+          const utxo = await utxoFromProcessorTransactionInput(
+            processor,
+            replacedTx,
+            index
+          )
+          // Mark as unspent because we're going to reuse replaced tx's UTXOs
+          utxo.spent = false
+          return utxo
+        })
+      )
+      // Get the largest input from the transaction to re-use:
+      const replacedTxMaxUtxo = replacedTxUtxos.reduce((maxUtxo, utxo) => {
+        if (utxo.value > maxUtxo.value) return utxo
+        return maxUtxo
+      })
+
+      const targets: MakeTxTarget[] = []
+      const newOurReceiveAddresses: string[] = []
+      let foundChangeAddress: string | undefined
+      for (const output of replacedTx.outputs) {
+        // Fetch address by output's scriptPubkey to determine output ownership
+        const ourAddress = await processor.fetchAddress(output.scriptPubkey)
+
+        // This isn't our output, so include it as a target
+        if (ourAddress == null) {
+          targets.push({
+            scriptPubkey: output.scriptPubkey,
+            value: parseInt(output.amount)
+          })
+          continue
+        }
+
+        // The output is ours:
+        // Addresses from database should include a path; this is a type assert
+        if (ourAddress.path != null) {
```

Comment:
If this fails, the output won't be in `newOurReceiveAddresses`, `foundChangeAddress`, or `targets`. Is this going to be a problem? Would it be safer to treat the address as an external target if this check fails?